### ISSUE TITLE
Unexpected } for SQUARE_OPEN

### ISSUE THREAD

**Issue by dirk-thomas** [bengardner/uncrustify#379]
The following code fails to parse:

    typedef std::vector<::std::basic_string<char, std::char_traits<char> > _names_type;

while this one (without the leading colons inside the template argument) works:

    typedef std::vector<std::basic_string<char, std::char_traits<char> > _names_type;

**Comment by mihaipopescu** [comment]
The same it's happening in our case:

```
jni::Array<::jbyte> stringbytes = java::lang::String(inputString).GetBytes("UTF-8");
```

I'll try to investigate.

**Comment by bengardner** [comment]
That's a trigraph issue. In C, <:: is another way to write {.
There might be a way to disable trigraphs, but I don't recall.  If not, there should be.

**Comment by mihaipopescu** [comment]
Indeed ! Actually there is a digraph `<:` that translates to `[`.

Digraphs are supported by Uncrustify but I don't see any option to disable them.
Trigraphs are not supported. Also the subject of removing them is apparently very debated in the C++ community.

https://en.wikipedia.org/wiki/Digraphs_and_trigraphs

**Comment by bengardner** [comment]
Should be fixed in 485eb91664716ffac58a49394e628e5e5db13af5.
I added the "enable_digraphs" option and defaulted it to false.

**Comment by mihaipopescu** [comment]
@bengardner I confirm the fix works. Thank you Ben !
